modules/apps/translation/translation-web/src/main/resources/META-INF/resources/js/translate/Translate.js

```diff
@@ -0,0 +1,320 @@
+		</ClayLayout.Col>
+	</ClayLayout.Row>
+);
+
+const Translate = ({
+	aditionalFields,
+	infoFieldSetEntries,
+	portletNamespace,
+	publishButtonDisabled,
+	publishButtonLabel,
+	redirectURL,
+	saveButtonDisabled,
+	saveButtonLabel,
+	sourceLanguageIdTitle,
+	targetLanguageIdTitle,
+	translateLanguagesSelectorData,
+	translationPermission,
+	updateTranslationPortletURL,
+	workflowActions,
+}) => {
+	const [formHaschanges, setFormHaschanges] = useState(false);
+	const [workflowAction, setWorkflowAction] = useState(
+		workflowActions.PUBLISH
+	);
+
+	return (
+		<form
+			action={updateTranslationPortletURL}
+			className="translation-edit"
+			method="POST"
+			name="translate_fm"
+		>
+			<input
+				defaultValue={workflowAction}
+				name={`${portletNamespace}workflowAction`}
+				type="hidden"
+			/>
+			{Object.entries(aditionalFields).map(([name, value]) => (
+				<input
+					defaultValue={value}
+					key={name}
+					name={`${portletNamespace}${name}`}
+					type="hidden"
+				/>
+			))}
+
+			<nav className="component-tbar subnav-tbar-light tbar">
+				<ClayLayout.ContainerFluid view>
+					<ul className="tbar-nav">
+						<li className="tbar-item tbar-item-expand">
+							<TranslateLanguagesSelector
+								{...translateLanguagesSelectorData}
+								formHaschanges={formHaschanges}
+								portletNamespace={portletNamespace}
+							/>
+						</li>
+						<li className="tbar-item">
+							<div className="metadata-type-button-row tbar-section text-right">
+								<ClayButton.Group spaced>
+									<ClayLink
+										button={{small: true}}
+										displayType="secondary"
+										href={redirectURL}
+									>
+										{Liferay.Language.get('cancel')}
+									</ClayLink>
+									<ClayButton
+										disabled={saveButtonDisabled}
+										displayType="secondary"
+										onClick={() => {
+											setWorkflowAction(
+												workflowActions.SAVE_DRAFT
+											);
+										}}
+										small
+										type="submit"
+									>
+										{saveButtonLabel}
+									</ClayButton>
+									<ClayButton
+										disabled={publishButtonDisabled}
+										displayType="primary"
+										small
+										type="submit"
+									>
+										{publishButtonLabel}
+									</ClayButton>
+								</ClayButton.Group>
+							</div>
+						</li>
+					</ul>
+				</ClayLayout.ContainerFluid>
+			</nav>
+			<ClayLayout.ContainerFluid view>
+				<div className="sheet translation-edit-body-form">
+					{!translationPermission ? (
+						<ClayAlert>
+							{Liferay.Language.get(
+								'you-do-not-have-permissions-to-translate-to-any-of-the-available-languages'
+							)}
+						</ClayAlert>
+					) : (
+						<>
+							<TranslationHeader
+								sourceLanguageIdTitle={sourceLanguageIdTitle}
+								targetLanguageIdTitle={targetLanguageIdTitle}
+							/>
+							<TranslationFieldSetEntries
+								infoFieldSetEntries={infoFieldSetEntries}
+								onChange={() => setFormHaschanges(() => true)}
```

Comment:
Is not the same `() => setFormHasChanges(true)`?

Reply:
yes, thx!